backend/server/src/db.rs

```diff
@@ -37,12 +37,31 @@ pub async fn create_pool(config: &config::Db) -> Result<Pool> {
         .context("failed to get DB connection")?;
     connection.execute("SELECT 1", &[]).await
         .context("failed to execute DB test query")?;
-    let n_roots = connection.execute("SELECT * from realms where id = 0", &[]).await
-        .context("failed to check")?;
-    if n_roots < 1 {
-        bail!("no root realm found");
-    } else if n_roots > 1 {
-        bail!("more than one root realm found");
+
+    // ensure database version exists
+    info!("Creating table version");
+    connection.execute(
+        "create table if not exists version (version int not null)", &[]).await
+        .context("failed to schema for table 'realms'")?;
+    if connection.query("SELECT count(*) < 1 from version", &[]).await?[0].get::<_, bool>(0) {
+        info!("Setting database version");
+        connection.execute("insert into version (version) values (0)", &[]).await
+            .context("failed to set database version")?;
+    }
+
+    // ensure table realms exist
+    info!("Creating table realms");
+    connection.execute("\
+        create table if not exists realms (\
+            id int generated always as identity (start with 0 minvalue 0) primary key,\
+            parent int not null references realms on delete restrict,\
+            name text not null\
+        )", &[]).await
+        .context("failed to schema for table 'realms'")?;
+    if connection.query("SELECT count(*) < 1 from realms where id = 0", &[]).await?[0].get::<_, bool>(0) {
+        info!("Inserting root realm");
+        connection.execute("insert into realms (name, parent) values ('root', 0)", &[]).await
+            .context("failed insert root realm")?;
```

Comment:
I think I'm against these kinds of "let's try to repair a strange DB state". Inserting the root realm is (should be) part of the corresponding migration. We can/should check this at some point, but if the root realm is not found, I would just error.